time_capture/scripts/employee.py

```diff
@@ -0,0 +1,23 @@
+import frappe
+from frappe import _
+
+
+def before_validate(doc, method):
+	validate_expected_working_hours(doc)
+
+
+def validate_expected_working_hours(doc):
+	if not doc.date_of_joining == min(ewh.valid_from for ewh in doc.expected_working_hours):
+		frappe.throw(_("Date of Joining is not the same as the earliest date of Expected Working Hours."))
+
+
+def get_expected_working_hours(employee_id, date):
+	"""
+	Get the expected working hours for an employee on a specific date.
+	"""
+	return frappe.db.get_value(
+		"Employee Expected Working Hours",
+		filters={"parent": employee_id, "valid_from": ["<=", date]},
```

Comment:
```suggestion
		filters={"parent": employee_id, "valid_from": ("<=", date)},
```

In general, favor tuples over lists where possible.